client/src/modules/cash/cash.service.js

```diff
@@ -123,60 +124,67 @@ function CashService(Modal, Api, Exchange, Session, moment, $translate) {
-      var value = params[column.field];
-
-      if (angular.isDefined(value)) {
-        column.value = value;
-
-        if (column.field === 'invoice_uuid') {
-          column.value = column.value.slice(0, LIMIT_UUID_LENGTH).concat('...');
-        }
-
-        return true;
-      } else {
-        return false;
-      }
-    });
+  cashFilters.registerDefaultFilters([
+    { key : 'period', label : 'TABLE.COLUMNS.PERIOD', valueFilter : 'translate' },
+    { key : 'custom_period_start', label : 'PERIODS.START', valueFilter : 'date' },
+    { key : 'custom_period_end', label : 'PERIODS.END', valueFilter : 'date' },
+    { key : 'limit', label : 'FORM.LABELS.LIMIT' }]);
+
+  cashFilters.registerCustomFilters([
+    { key : 'is_caution', label : 'FORM.LABELS.CAUTION' },
+    { key : 'cashbox_id', label : 'FORM.LABELS.CASHBOX' },
+    { key : 'debtor_uuid', label : 'FORM.LABELS.CLIENT' },
+    { key : 'user_id', label : 'FORM.LABELS.USER' },
+    { key : 'reference', label : 'FORM.LABELS.REFERENCE' },
+    { key : 'dateFrom', label : 'FORM.LABELS.DATE_FROM', comparitor : '>', valueFilter : 'date' },
+    { key : 'dateTo', label : 'FORM.LABELS.DATE_TO', comparitor : '<', valueFilter : 'date' },
+    { key : 'currency_id', label : 'FORM.LABELS.CURRENCY' },
+    { key : 'reversed', label : 'CASH.REGISTRY.REVERSED_RECORDS' },
+    { key : 'patientReference', label : 'FORM.LABELS.REFERENCE_PATIENT' },
+    { key : 'defaultPeriod', label : 'TABLE.COLUMNS.PERIOD', valueFilter : 'translate' },
+    { key : 'invoiceReference', label : 'FORM.LABELS.INVOICE' },
+    { key : 'patientReference', label : 'FORM.LABELS.REFERENCE_PATIENT' },
+    { key : 'invoice_uuid', label : 'FORM.LABELS.INVOICE' }]);
+
+  if (filterCache.filters) {
+    // load cached filter definition if it exists
+    cashFilters.loadCache(filterCache.filters);
   }
 
-  //open a dialog box to Cancel Cash Payment
-  function openCancelCashModal(invoice) {
-    return Modal.open({
-      templateUrl : 'modules/cash/modals/modal-cancel-cash.html',
-      resolve     : { data: { invoice: invoice } },
-      size        : 'md',
-      animation   : false,
-      keyboard    : false,
-      backdrop    : 'static',
-      controller  : 'ModalCancelCashController as ModalCtrl',
-    }).result;
+  // once the cache has been loaded - ensure that default filters are provided appropriate values
+  assignDefaultFilters();
+
+  function assignDefaultFilters() {
+    // get the keys of filters already assigned - on initial load this will be empty
+    var assignedKeys = Object.keys(cashFilters.formatHTTP());
+
+    // assign default period filter
+    var periodDefined =
+      service.util.arrayIncludes(assignedKeys, ['period', 'custom_period_start', 'custom_period_end']);
+
+    if (!periodDefined) {
+      cashFilters.assignFilters(Periods.defaultFilters());
+    }
+
+    // assign default limit filter
+    if (assignedKeys.indexOf('limit') === -1) {
+      cashFilters.assignFilter('limit', 100);
```

Comment:
It would be good to define a system wide default for some of these filters so that every module uses the same value (potentially through `bhConstants`). We can create an issue tracking the best practice for this.